hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleAccountState.java

```diff
@@ -416,6 +429,11 @@ public void setMemo(String memo) {
 		this.memo = memo;
 	}
 
+	public void setTransactionCounter(long transactionCounter) {
+		assertMutable("transactionCounter");
```

Comment:
Here also, shouldn't we be able to update the `transactionCounter` on immutable accounts?